src/synthesized_datasets/_datasets.py

```diff
@@ -55,34 +70,80 @@ def load(self) -> _pd.DataFrame:
             df = _pd.read_parquet(self.url)
         else:
             # CSV load is the default
-            df = _pd.read_csv(self.url)
+            dtypes = {
+                col: (
+                    dtype.value
+                    if dtype not in [_DType.DATETIME, _DType.TIMEDELTA]
+                    else "string"
+                )
+                for col, dtype in self._schema.items()
+            }
+            df = _pd.read_csv(self.url, dtype=dtypes)
+            for col, dtype in self._schema.items():
+                if dtype is _DType.DATETIME:
+                    df[col] = _pd.to_datetime(df[col], dayfirst=True)
+                if dtype is _DType.TIMEDELTA:
+                    df[col] = _pd.to_timedelta(df[col])
         df.attrs["name"] = self.name
         return df
 
-    def load_spark(self, spark: _typing.Optional[_ps.SparkSession] = None) -> _ps.DataFrame:
+    def load_spark(self, spark: _ty.Optional[_ps.SparkSession] = None) -> _ps.DataFrame:
         """Loads the dataset as a Spark DataFrame."""
 
         if spark is None:
             spark = _ps.SparkSession.builder.getOrCreate()
 
+        schema = _create_pyspark_schema(self._schema)
         spark.sparkContext.addFile(self.url)
         _, filename = _os.path.split(self.url)
         if self.url.endswith("parquet"):
             df = spark.read.parquet(_SparkFiles.get(filename))
         else:
             # CSV load is the default
-            df = spark.read.csv(_SparkFiles.get(filename), header=True, inferSchema=True)
+            df = spark.read.csv(
+                _SparkFiles.get(filename),
+                header=True,
+                schema=schema,
+                enforceSchema=False,
```

Comment:
should this be True?

Reply:
haha "Though the default value is true, it is recommended to disable the enforceSchema option to avoid incorrect results." 

Reading the above I was thinking it should be True until I got to this sentence